internal/services/detector.go

```
@@ -27,78 +28,139 @@ type Detector interface {
 	return &detector{
 		client: client,
 	}
 }
 
-// Detect detects the Linux distribution on the connected remote system.
+// Global atomic metrics for distribution detection performance
+var (
+	detectionAttempts  atomic.Uint64 // Total detection attempts
+	detectionSuccesses atomic.Uint64 // Successful detections
+	detectionFailures  atomic.Uint64 // Failed detections
+	cacheHits          atomic.Uint64 // Cache hits for performance
```

golangci-lint (1.64.8)

var `cacheHits` is unused

(unused)